tools/scripts/generic.mk

```diff
@@ -0,0 +1,13 @@
+ifeq (aducm3029,$(strip $(PLATFORM)))
```

Comment:
We should either make every makefile solve a certain platform independent of operating system (Example: aducm.mk, xilinx.mk, altera.mk, etc), or make every makefile solve for operating system, regardless of platform (Example: leaving windows.mk and linux.mk and including aducm.mk in them).

Reply:
I think it would be easier to do the latter of the two, so I tend to lean towards that.

Reply:
I rater use a OS agnostic makefile. If some changes are needed for one platform then should be changed in one place.
Also each new feature should be implemented in 2 different files.
My opinion is to make this generic.mk OS agnostic that handles all common stuff for each platform and for platform dependent stuff use aducm.mk, xilinx.mk, altera.mk and so on.

Reply:
Can we do this while preserving backwards compatibility?

Reply:
The current PR is backward compatible. When generic.mk will be used in all the projects, then we can refactor these files